.circleci/config.yml

```diff
@@ -285,234 +282,101 @@ commands:
+  version: 2.1
   regular:
     jobs:
-      - unit_tests_36
-      - linters_36
-      - e2e_tests_36
-      - docs_36
-      - docs_linkcheck_37
-      - unit_tests_37
-      - linters_37
-      - e2e_tests_37
-      - docs_37
-      - unit_tests_38
-      - linters_38
-      - e2e_tests_38
-      - pip_compile_36
-      - pip_compile_37
-      - pip_compile_38
-      - win_unit_tests_36
-      - win_unit_tests_37
-      - win_unit_tests_38
-      - win_pip_compile_36
-      - win_pip_compile_37
-      - win_pip_compile_38
-      - win_e2e_tests_36
-      - win_e2e_tests_37
-      - win_e2e_tests_38
-      - run_kedro_viz:
+      - e2e_tests:
+          matrix:
+            parameters:
+              python_version: ["3.6", "3.7", "3.8"]
+      - win_e2e_tests:
+          matrix:
+            parameters:
+              python_version: ["3.6", "3.7", "3.8"]
+      - unit_tests:
+          matrix:
+            parameters:
+              python_version: ["3.6", "3.7", "3.8"]
+      - win_unit_tests:
+          matrix:
+            parameters:
+              python_version: ["3.6", "3.7", "3.8"]
+      - lint:
+          matrix:
+            parameters:
+              python_version: ["3.6", "3.7", "3.8"]
+      - pip_compile:
+          matrix:
+            parameters:
+              python_version: ["3.6", "3.7", "3.8"]
+      - win_pip_compile:
+          matrix:
+            parameters:
+              python_version: ["3.6", "3.7", "3.8"]
+      - build_docs
+      - docs_linkcheck
+      - viz_build:
           filters:
             branches:
-              only:
-                - master
+              only: main
+
       - all_circleci_checks_succeeded:
           requires:
-            - unit_tests_36
-            - linters_36
-            - e2e_tests_36
-            - docs_36
-            - unit_tests_37
-            - linters_37
-            - e2e_tests_37
-            - docs_37
-            - docs_linkcheck_37
-            - unit_tests_38
-            - linters_38
-            - e2e_tests_38
-            - pip_compile_36
-            - pip_compile_37
-            - pip_compile_38
-            - win_pip_compile_36
-            - win_pip_compile_37
-            - win_pip_compile_38
-            - win_unit_tests_36
-            - win_unit_tests_37
-            # Skipped due to `pywin32 is in an unsupported or invalid wheel`
-            # - win_e2e_tests_36
-            # Skipped due to Windows fatal exception: stack overflow
-            # - win_unit_tests_38
-            - win_e2e_tests_37
-            - win_e2e_tests_38
+            # Should just have e2e_tests and win_e2e_tests but we need to skip
+            # win_e2e_tests-3.6 due to `pywin32 is in an unsupported or invalid wheel`
+            - e2e_tests
+            - win_e2e_tests-3.7
+            - win_e2e_tests-3.8
+            # Should just have unit_tests and win_unit_tests but we need to skip
+            # win_unit_tests-3.8 due to `Windows fatal exception: stack overflow`
```

Comment:
We can put back `win_e2e_tests-3.6` as @limdauto fixed them. And actually I haven't seen the windows unit test failure on 3.8 in a long time, so I'm happy to have both back in, and simplify this to be
```yaml
- unit_tests
- win_unit_tests
- win_e2e_tests
```

Reply:
Awesome! Done in 005ef88 - much tidier now.